### ISSUE TITLE
beatmap downloads truncated to 1MB?

### ISSUE THREAD

**Issue by Tillerino** [ppy/osu-api#131]
Well this is technically not an "API" issue, but I guess this place is as good as any to post about this.

It looks like beatmap files are truncated when downloaded via the website, e.g. `https://osu.ppy.sh/osu/777867`. This download is truncated to _exactly_ 1MB, so it's probably no coincidence.

There are a couple of more beatmaps where this occurrs:

```
893209
621903
621905
621907
871542
777875
616659
```

**Comment by peppy** [comment]
Honestly not sure we should be ever promoting .osu files over 1mb as being a usable thing...

**Comment by omkelderman** [comment]
b-b-but... awesome storyboards?

**Comment by hypereddie** [comment]
> Honestly not sure we should be ever promoting .osu files over 1mb as being a usable thing...
The maps shouldn't be accepted then. Accepting the maps and truncating them seems weird IMO...

**Comment by Tillerino** [comment]
@peppy Any updates? Been a year...

**Comment by peppy** [comment]
I just looked into this today. The issue is quite deeply ingrained in BSS. I've added a todo for fixing it and estimate it will be complete this week.

tl;dr the server-side files are actually truncated and need to be repaired.

**Comment by Syriiin** [comment]
Looks like it's been fixed recently. Time to close this ancient issue?

**Comment by peppy** [comment]
Yep, has since been fixed, and I believe we've retroactively fixed all the broken files.
